src/main/resources/hudson/plugins/promoted_builds/PromotionProcess/process-config.jelly

```diff
@@ -37,6 +37,19 @@
         </select>
       </f:entry>
 
+      <j:set var="jdks" value="${app.JDKs}" />
+      <f:advanced>
+        <f:entry title="JDK"
+                 description="${%JDK to be used for this project}">
+          <select class="setting-input validated" name="jdkForPromotion" checkUrl="'${rootURL}/defaultJDKCheck?value='+this.value">
+            <f:option selected="${(inst == null) || (inst.name=='UseOwner')}" value="UseOwner">${%UseSameAsJob}</f:option>
```

Comment:
Probably makes sense to add a help file to describe the behavior and the magic <code>UseOwner</code> string. A common way would be to just use empty/null string to specify this default behavior.